src/org/thoughtcrime/securesms/profiles/AvatarHelper.java

```diff
@@ -1,95 +1,58 @@
 package org.thoughtcrime.securesms.profiles;
 
 
-import android.annotation.SuppressLint;
 import android.content.Context;
 import android.graphics.Bitmap;
+
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
 import com.b44t.messenger.DcContext;
 
 import org.thoughtcrime.securesms.connect.DcHelper;
-import org.thoughtcrime.securesms.database.Address;
 
 import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
-import java.io.InputStream;
 
 public class AvatarHelper {
 
-    private static final String GROUP_TEMPLATE = "group_chat_avatar_%s_%o.jpg";
-
-    private static final String AVATAR_DIRECTORY = "avatars";
-
-    @SuppressLint("DefaultLocale")
-    private static String getFilePathForGroupAvatar(Context context, int chatId, long timestamp) {
-        return getAvatarDirectoryPath(context) + String.format(GROUP_TEMPLATE, chatId, timestamp);
-    }
-
-    @NonNull
-    private static String getAvatarDirectoryPath(Context context) {
-        return context.getFilesDir().getAbsolutePath() + File.separator + AVATAR_DIRECTORY + File.separator;
-    }
-
     public static void setGroupAvatar(Context context, int chatId, Bitmap bitmap) {
-        if (bitmap == null) {
-            return;
-        }
-        //noinspection ResultOfMethodCallIgnored
-        new File(getAvatarDirectoryPath(context)).mkdirs();
         DcContext dcContext = DcHelper.getContext(context);
-        String avatarPath = dcContext.getChat(chatId).getProfileImage();
-        if (avatarPath != null && !avatarPath.isEmpty()) {
-            File oldImage = new File(avatarPath);
-            if (oldImage.exists()) {
+
+        if (bitmap == null) {
+            dcContext.setChatProfileImage(chatId, null);
+        } else {
+            try {
+                File avatar = File.createTempFile("groupavatar", "jpg", context.getCacheDir());
```

Comment:
the extension needs to be `.jpg` - otherwise, at least the control showing it has problems.